web-registry/src/features/ecocredit/CreateBatchBySteps/CreateBatchMultiStepForm/CreditBasics.tsx

```diff
@@ -87,79 +90,77 @@ export const validationSchemaFields = {
 export const validationSchema = Yup.object(validationSchemaFields);
 
 export const initialValues = {
-  classId: '',
+  projectId: '',
   startDate: null,
   endDate: null,
   metadata: '',
 };
 
 type Props = {
-  saveCreditClassSelected: (creditClass: Option) => void;
-  saveProjectSelected: (project: Option) => void;
+  saveProjectOptionSelected: (project: Option) => void;
 };
 
 export default function CreditBasics({
-  saveCreditClassSelected,
-  saveProjectSelected,
+  saveProjectOptionSelected,
 }: Props): React.ReactElement {
+  const { wallet } = useWallet();
+  const { data: projects } = useQueryProjectsByIssuer(wallet!.address);
+
   const { values, validateForm } = useFormikContext<CreditBasicsFormValues>();
-  const metadata = values.metadata as VCSBatchMetadataLD;
-  const projectId = metadata['regen:vcsProjectId'];
-  // TODO: this is a hack to make V4 testnet work. C02 is a copy of C01 on V4. In PROD/mainnet, this should only be C01.
-  const isVCS = values.classId === 'C01' || values.classId === 'C02';
 
-  // to store on partial submit the selected credit class option,
-  // and the project option in order to complete display name in Review step
-  const [creditClassOptions, setCreditClassOptions] =
-    React.useState<Option[]>();
-  const [projectOptions, setProjectOptions] = React.useState<Option[]>();
+  const [projectOptions, setProjectOptions] = useState<Option[]>([]);
+  const [classId, setClassId] = useState<ClassID>();
+  const [isVCS, setIsVCS] = useState<boolean>();
 
-  React.useEffect(() => {
-    const isFound = creditClassOptions?.find(
-      item => item.value === values.classId,
-    );
-    if (isFound) saveCreditClassSelected(isFound);
-  }, [values.classId, creditClassOptions, saveCreditClassSelected]);
+  const { projectId } = values;
 
-  React.useEffect(() => {
+  useEffect(() => {
```

Comment:
It's nice to have logic split into separate effects 👍 
It would be even nicer to have them extracted as hooks.
This would allow to:
- name them (ie: `useUpdateProjectClass` / `useUpdateProjectOptions` / `useSaveProjectSelectedOption`) -> help to understanding what the code is about
- make the file a little smaller (since it's already ~300 Loc)

What do you think ?

Reply:
super good idea, totally agree! Thanks!